release/default.nix

```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+{ pkgs ? import (import ../nix/sources.nix).nixpkgs { } }:
+let
+  executable = (import ../crossref-verifier.nix {
+    static = true;
+  }).components.exes.crossref-verify;
+  joinByBasename = name: paths:
+    pkgs.runCommandNoCC name { } (''
+      mkdir $out
+    '' + pkgs.lib.concatMapStrings (path: ''
+      ln -s ${path} $out/${builtins.baseNameOf path}
+    '') paths);
+in joinByBasename "crossref-verifier-release" [
+  "${executable}/bin"
```

Comment:
As I understand, the static binary is needed to make it easy to run crossref-verifier from another CI, so how do I do it? Also, `./default.nix` is not needed anymore as well as support for building a dynamic executable.

Reply:
To run a static version,
```
wget https://github.com/serokell/crossref-verifier/releases/download/auto-release/release.tar.gz
tar -xf release.tar.gz
./crossref-verifier/bin/crossref-verify
```

And I believe it's better to build a dynamic executable for regular CI because it should generally be faster due to better binary cache support for it.

Reply:
What do you mean by binary cache support?

Reply:
I mean there are no pre-built libraries and compilers for musl upstream.

Reply:
In most cases they will rebuild only when we repin nixpkgs, and the effect is negligible compared to the fact that we have to rebuild ghc and all haskell packages as well. But it would be simpler for us to have to maintain the static version only